lib/modules/datasource/galaxy-collection/index.ts

```diff
@@ -35,91 +31,89 @@ export class GalaxyCollectionDatasource extends Datasource {
   }: GetReleasesConfig): Promise<ReleaseResult | null> {
     const [namespace, projectName] = packageName.split('.');
 
-    // TODO: types (#22198)
-    const baseUrl = `${registryUrl}api/v2/collections/${namespace}/${projectName}/`;
-
-    let baseUrlResponse: HttpResponse<BaseProjectResult>;
-    try {
-      baseUrlResponse = await this.http.getJson<BaseProjectResult>(baseUrl);
-    } catch (err) {
-      this.handleGenericErrors(err);
-    }
+    const baseUrl = ensureTrailingSlash(
+      joinUrlParts(
+        registryUrl!,
+        'api/v3/plugin/ansible/content/published/collections/index',
+        namespace,
+        projectName,
+      ),
+    );
 
-    if (!baseUrlResponse?.body) {
-      logger.warn(
-        { dependency: packageName },
-        `Received invalid data from ${baseUrl}`,
-      );
-      return null;
+    const { val: baseProject, err: baseErr } = await this.http
+      .getJsonSafe(baseUrl, GalaxyV3)
+      .onError((err) => {
+        logger.warn(
+          { datasource: this.id, packageName, err },
+          `Error fetching ${baseUrl}`,
+        );
+      })
+      .unwrap();
+    if (baseErr) {
+      this.handleGenericErrors(baseErr);
     }
 
-    const baseProject = baseUrlResponse.body;
-
-    const versionsUrl = `${baseUrl}versions/`;
-
-    let versionsUrlResponse: HttpResponse<VersionsProjectResult>;
-    try {
-      versionsUrlResponse =
-        await this.http.getJson<VersionsProjectResult>(versionsUrl);
-    } catch (err) {
-      this.handleGenericErrors(err);
+    const versionsUrl = ensureTrailingSlash(joinUrlParts(baseUrl, 'versions'));
+
+    const { val: versionsProject, err: versionsErr } = await this.http
+      .getJsonSafe(versionsUrl, GalaxyV3Versions)
+      .onError((err) => {
+        logger.warn(
+          { datasource: this.id, packageName, err },
+          `Error fetching ${versionsUrl}`,
+        );
+      })
+      .unwrap();
+    if (versionsErr) {
+      this.handleGenericErrors(versionsErr);
     }
 
-    const versionsProject = versionsUrlResponse.body;
-
-    const releases: Release[] = versionsProject.results.map((value) => {
+    const releases = versionsProject.data.map((value) => {
       const release: Release = {
         version: value.version,
+        releaseTimestamp: value.created_at,
         isDeprecated: baseProject.deprecated,
       };
       return release;
     });
```

Comment:
why not doing this as a schema transform? Like we do on other datasources and managers?

Reply:
If find this more readable TBH. Tough if we want to keep this as a new convention we should add this to the dev docs